IntegrationAdaptersTests/RegisterHospitalTests.cs

```diff
@@ -29,17 +29,51 @@ public void Is_pharmacy_already_existing_on_hospital()
         [Fact]
         public void Is_pharmacy_not_existing_on_hospital()
         {
-            var options = new DbContextOptionsBuilder<HealthCareSystemDbContext>()
-                .UseMySql(connectionString: "server=localhost;port=3306;database=newmydb;user=root;password=root")
-                .Options;
+            ApiService service = new ApiService();
 
-            ApiService service = new ApiService(new HealthCareSystemDbContext(options));
-
-            Api a = new Api("m", "m", "m");
+            Api a = new Api("l", "l", "l");
 
             bool notExists = service.RegisterHospitalOnPharmacy(a);
 
             Assert.True(notExists);
         }
+        [Fact]
+        public void Is_apiKey_null()
+        {
+            mock = new Mock<IApiRepository>();
+            mock.Setup(expression:t => t.GetAllApis()).Returns(new List<Api> { new Api { Key = null, PharmacyName = "name1", Url = "url1" }});
+            ApiService service = new ApiService();
+            Api api = mock.Object.GetAllApis()[0];
+            Assert.Null(service.getApiKey(api));
+        }
+        [Fact]
+        public void Is_apiKey_not_null()
+        {
+            mock = new Mock<IApiRepository>();
+            mock.Setup(expression: t => t.GetAllApis()).Returns(new List<Api> { new Api { Key = "kljuc1", PharmacyName = "name1", Url = "url1" } });
+            ApiService service = new ApiService();
+            Api api = mock.Object.GetAllApis()[0];
+            Assert.NotNull(service.getApiKey(api));
+        }
```

Comment:
Is there a business logic case that requires this check?

Reply:
Formatting tip: Add an empty row between test functions so that they are easier to find 😄 I also like to add an empty line between arrange, act and assert sections to make the test more readable